src/schedule.py

```diff
@@ -310,6 +310,62 @@ def log_register_use(computation_graph, step, hw_element_counts, execution_time)
     keys.sort()
     in_use_sorted = {i: in_use[i] for i in keys}
     write_df(in_use_sorted, hw_element_counts, execution_time, step)
+
+def check_valid_hw(computation_graph, hw_netlist):
+    """
+    Checks if every operatory node if bi-directionally connected to every register node.
+    """
+    hw_graph = nx.DiGraph()
+    
+    for node_id, attrs in hw_netlist['nodes'].items():
+        hw_graph.add_node(node_id, **attrs)
+    
+    for edge in hw_netlist['edges']:
+        hw_graph.add_edge(*edge)
+    
+    register_nodes = [node for node, data in hw_graph.nodes(data=True) 
+                      if data.get('type') == 'memory' and data.get('function') == 'Regs']
+    
+    operator_nodes = [node for node, data in hw_graph.nodes(data=True) 
+                      if data.get('type') == 'pe']
+    
+    for register in register_nodes:
+        all_bidirectional = True
+        for operator in operator_nodes:
+            if not (hw_graph.has_edge(operator, register) and hw_graph.has_edge(register, operator)):
+                all_bidirectional = False
+                break
+        if all_bidirectional:
+            return True
```

Comment:
You don't need a given register to be connected to all operators.
you need a given operator to be connected to all other operators.
eg. (slight variation of what I slacked you)
eg nodes: mult, add, sub, reg1, reg2, reg3
edges: mult - reg1 - add
mult - reg2 - sub
add - reg3 - sub
is an allowed arch